### DISCUSSION TITLE
How does Renovate handle Composer updates?

### DISCUSSION THREAD

**Discussion by felixranesberger** [renovatebot/renovate#23374] [Request Help]
### How are you running Renovate?

Development mode - Docker run command

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

Never worked

### Wanted end result.

related pull request: https://github.com/renovatebot/renovate/pull/23372/files
related issue: https://github.com/renovatebot/renovate/issues/6295

I'm working on a pull request that allows to specify additional options to the `composer update` command that Renovate uses to provide the pull requests.

For some reason, the Renovate update result is different, than if I run the same command myself in the terminal.

I got the command that Renovate used from the debug log. 

(The `--with-all-dependencies comes from my pull request)`

```bash
composer update typo3/cms-adminpanel typo3/cms-backend typo3/cms-belog typo3/cms-beuser typo3/cms-core typo3/cms-dashboard typo3/cms-extbase typo3/cms-extensionmanager typo3/cms-filelist typo3/cms-filemetadata typo3/cms-fluid typo3/cms-fluid-styled-content typo3/cms-frontend typo3/cms-info typo3/cms-install typo3/cms-lowlevel typo3/cms-recordlist typo3/cms-recycler typo3/cms-rte-ckeditor typo3/cms-scheduler typo3/cms-setup typo3/cms-sys-note typo3/cms-tstemplate typo3/cms-viewpage --with-dependencies --ignore-platform-req='ext-*' --ignore-platform-req='lib-*' --with-all-dependencies --no-ansi --no-interaction --no-scripts --no-autoloader --no-plugins
```

### What you tried so far.

I tried using the same command to make the composer update locally and got different results:

**Result with Renovate:**
https://github.com/felixranesberger/renovate-testing-repo-2/pull/7/files

**Result with manual update:**
https://github.com/felixranesberger/renovate-testing-repo-2/compare/main...typo3-update-manual-2?diff=split

### Relevant debug logs

Command used to run Renovate in debug mode:

```bash
docker run -it --rm -v "$PWD":/usr/src/app -e LOG_LEVEL=debug -e RENOVATE_TOKEN=<renovate-token> -e RENOVATE_COMPOSER_ADDITIONAL_ARGUMENTS="--with-all-dependencies" -w /usr/src/app renovatebot_local yarn start felixranesberger/renovate-testing-repo-2 > renovate-ou
tput.txt
```

**Comment by viceice** [comment]
different composer/ PHP versions?

**Comment by viceice** [comment]
please create a discussion about the new feature before opening a PR, so we can agree about the implementation details.
